packages/rainbowkit/src/wallets/walletConnectors/enkryptWallet/enkryptWallet.ts

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import { ConnectorNotFoundError, Ethereum } from '@wagmi/core';
+import { getAddress } from 'ethers/lib/utils';
+import { InjectedConnector } from 'wagmi/connectors/injected';
+import { Chain } from '../../../components/RainbowKitProvider/RainbowKitChainContext';
+import { Wallet } from '../../Wallet';
+
+export interface EnkryptWalletOptions {
+  chains: Chain[];
+  shimDisconnect?: boolean;
+}
+
+interface EIP5749ConnectorOptions {
+  walletId: string;
+  chains: Chain[];
+  options: {
+    shimDisconnect: boolean | undefined;
+  };
+}
+const detectProvider = (
+  walletId: string
+): {
+  isAvailable: boolean;
+  provider?: Ethereum;
+} => {
+  const isAvailable =
+    typeof window !== 'undefined' &&
+    typeof (window as any).evmproviders !== 'undefined' &&
+    (window as any).evmproviders[walletId]
+      ? true
+      : false;
+  if (isAvailable)
+    return {
+      isAvailable,
+      provider: (window as any).evmproviders[walletId],
+    };
+  return { isAvailable };
+};
+class EIP5749Connector extends InjectedConnector {
+  walletId: string;
```

Comment:
Does Enkrypt currently support `window.ethereum`, an alternate `window.wallet`, or `window.ethereum.providers`? Would like to see 5749 settled and introduced in Wagmi before we adopt, and would prefer to adopt the current standards in this connector. We will migrate all connectors to 5749 once that has moved forward.

Reply:
it does, however is there any specific issue regarding this approach?

Reply:
@kvhnuke Only concerned about fragility. RainbowKit inherits `window.ethereum.providers` support from Wagmi. Because Enkrypt is currently unique, we would like to adopt the connector through `@wagmi/connectors` akin to the `MetaMaskConnector` or `InjectedConnector` implementations there. We can then inherit that upstream in RainbowKit for the detection logic.

Reply:
but I see Phantom is using `window.phantom.ethereum`? will it be ok to use `window.enkrypt.providers.ethereum`?